firstvoices/backend/models/widget.py

```diff
@@ -0,0 +1,149 @@
+    default = 0
+    left = 10
+    right = 20
+
+
+class Widget(BaseModel):
+    class Meta:
+        verbose_name = _("widget")
+        verbose_name_plural = _("widgets")
+        rules_permissions = {
+            "view": rules.always_allow,
+            "add": predicates.is_superadmin,
+            "change": predicates.is_superadmin,
+            "delete": predicates.is_superadmin,
+        }
+
+    title = models.CharField(max_length=225)
+    type = models.IntegerField(
+        choices=WidgetTypes.choices, default=WidgetTypes.WIDGET_TEXT
+    )
+    format = models.IntegerField(
+        choices=WidgetFormats.choices, default=WidgetFormats.default
+    )
+
+    def __str__(self):
+        return self.title
+
+
+class WidgetSettings(BaseModel):
+    class Meta:
+        verbose_name = _("widget setting")
+        verbose_name_plural = _("widget settings")
+        rules_permissions = {
+            "view": rules.always_allow,
+            "add": predicates.is_superadmin,
+            "change": predicates.is_superadmin,
+            "delete": predicates.is_superadmin,
+        }
+
+    widget = models.ForeignKey(
+        Widget, on_delete=models.CASCADE, related_name="%(class)s_set"
+    )
+    key = models.CharField(max_length=225)
+    value = models.TextField()
+
+    def __str__(self):
+        return self.key
+
+
+class SiteWidget(Widget, BaseControlledSiteContentModel):
+    class Meta:
+        verbose_name = _("site widget")
+        verbose_name_plural = _("site widgets")
+        rules_permissions = {
+            "view": predicates.is_visible_object,
+            "add": predicates.is_superadmin,  # permissions will change when we add a write API
+            "change": predicates.is_superadmin,
+            "delete": predicates.is_superadmin,
+        }
+
+    def __str__(self):
+        return self.title
+
+
+class SiteWidgetList(BaseSiteContentModel):
+    class Meta:
+        verbose_name = _("site widget list")
+        verbose_name_plural = _("site widget lists")
+        rules_permissions = {
+            "view": predicates.has_visible_site,
+            "add": predicates.is_superadmin,  # permissions will change when we add a write API
+            "change": predicates.is_superadmin,
+            "delete": predicates.is_superadmin,
+        }
+
+    widgets = models.ManyToManyField(
+        SiteWidget, related_name="sitewidget_set", through="SiteWidgetListOrder"
+    )
+    title = models.CharField(max_length=225)
+
+    def __str__(self):
+        return f"{self.title} - [{self.site}]"
+
+
+class SiteWidgetListOrder(BaseModel):
+    class Meta:
+        verbose_name = _("site widget list order")
+        verbose_name_plural = _("site widget list orders")
+        constraints = [
+            models.UniqueConstraint(
+                fields=["order", "site_widget_list"], name="unique_widget_order"
+            ),
+        ]
+        rules_permissions = {
+            "view": rules.always_allow,
```

Comment:
I think this should be has_visible_site

Reply:
As discussed in a call we will leave this for now and investigate other options in FW-4368. An example of how to deal with permissions has been left in an example branch: https://github.com/First-Peoples-Cultural-Council/fv-be/blob/457043bc6902e3b0c8c0a37db4224db1f670891e/firstvoices/backend/models/widget.py#L76-L133